### PR TITLE
feat: add insert intent management command

### PR DESCRIPTION
**Pull request by alangsto**
**JIRA:** [MST-1495](https://2u-internal.atlassian.net/browse/MST-1495)

**Description:** This management command requires a CSV file containing data for various program intents, and will then create program intent objects based on that data. 

If a program intent defined in the CSV file is a duplicate, a new intent object will not be created. 

**Merge checklist:**
- [ ] All reviewers approved
- [x] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)

### CODE REVIEW

**Review by MichaelRoytman — APPROVED** [review]
This looks straightforward to me. I left one comment about the `DictReader` class.

File: program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+                existing_intent = ProgramIntent.objects.filter(
+                    program_uuid=program_uuid,
+                    lms_user_id=lms_user_id,
+                    reason=intent_reason,
+                    certainty=certainty,
+                    effective_timestamp=effective_time
+                ).first()
+
+                if not existing_intent:
```
Comment (ashultz0) [inline program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py:71]: can we use one of the or-create methods to make this simpler?

https://docs.djangoproject.com/en/dev/ref/models/querysets/#update-or-create

Comment (alangsto) [reply program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py:71]: Oh yeah! Looks like `get_or_create` would be a good candidate here

Comment (alangsto) [reply program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py:71]: Ah wait I remember why I didn't do this, I wanted to be able to bulk create attempts, which I would not be able to do via one of the `or-create` methods.

Comment (ashultz0) [reply program_intent_engagement/apps/core/management/commands/bulk_insert_intents.py:71]: oh those might be even more efficient 👍 

you might be able to bulk query out the possible existing ones but that might be too fat a query for django

**Pull request merged**